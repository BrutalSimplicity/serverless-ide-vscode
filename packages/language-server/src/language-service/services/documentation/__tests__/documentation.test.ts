import { DocumentationService } from './../index';

const docUrl =
	'https://d1uauaxba7bl26.cloudfront.net/latest/gzip/CloudFormationResourceSpecification.json';

describe('DocumentationService', () => {
	let service: DocumentationService = void 0;

	beforeEach(() => {
		service = new DocumentationService(docUrl);
	});

	test('should return undefined if documentation source is not found', async () => {
		const brokenService = new DocumentationService('unknown');

		const resourceDoc = await brokenService.getResourceDocumentation(
			'AWS::DynamoDB::Table'
		);
		const propertyDoc = await brokenService.getPropertyDocumentation(
			'AWS::DynamoDB::Table',
			'KeySchema'
		);

		expect(resourceDoc).toBeUndefined();
		expect(propertyDoc).toBeUndefined();
	});

	test('return dynamodb resource documentation', async () => {
		const docs = await service.getResourceDocumentation(
			'AWS::DynamoDB::Table'
		);

		expect(docs).toMatchSnapshot();
	});

	test('should return undefined for non-existing resource', async () => {
		const doc = await service.getResourceDocumentation('unknown');

		expect(doc).toBeUndefined();
	});

	test('should return dynamodb resource documentation', async () => {
		const docs = await service.getPropertyDocumentation(
			'AWS::DynamoDB::Table',
			'KeySchema'
		);

		expect(docs).toMatchSnapshot();
	});

	test('should return undefined if property documentation is not found', async () => {
		const docs1 = await service.getPropertyDocumentation(
			'unknown',
			'KeySchema'
		);

		const docs2 = await service.getPropertyDocumentation(
			'AWS::DynamoDB::Table',
			'unknown'
		);

		expect(docs1).toBeUndefined();
		expect(docs2).toBeUndefined();
	});
});
