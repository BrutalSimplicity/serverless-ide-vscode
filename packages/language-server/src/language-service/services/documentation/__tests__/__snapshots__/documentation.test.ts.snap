// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentationService e2e return dynamodb resource documentation 1`] = `
"[AWS::DynamoDB::Table](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dynamodb-table.html) / Properties


-----
- [AttributeDefinitions](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dynamodb-table.html#cfn-dynamodb-table-attributedef): [ AttributeDefinition ]
- [BillingMode](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dynamodb-table.html#cfn-dynamodb-table-billingmode): String
- [GlobalSecondaryIndexes](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dynamodb-table.html#cfn-dynamodb-table-gsi): [ GlobalSecondaryIndex ]
- [KeySchema](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dynamodb-table.html#cfn-dynamodb-table-keyschema): [ [KeySchema](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-dynamodb-keyschema.html) ]
- [LocalSecondaryIndexes](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dynamodb-table.html#cfn-dynamodb-table-lsi): [ LocalSecondaryIndex ]
- [PointInTimeRecoverySpecification](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dynamodb-table.html#cfn-dynamodb-table-pointintimerecoveryspecification): [PointInTimeRecoverySpecification](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-dynamodb-table-pointintimerecoveryspecification.html)
- [ProvisionedThroughput](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dynamodb-table.html#cfn-dynamodb-table-provisionedthroughput): [ProvisionedThroughput](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-dynamodb-provisionedthroughput.html)
- [SSESpecification](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dynamodb-table.html#cfn-dynamodb-table-ssespecification): [SSESpecification](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-dynamodb-table-ssespecification.html)
- [StreamSpecification](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dynamodb-table.html#cfn-dynamodb-table-streamspecification): [StreamSpecification](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-dynamodb-streamspecification.html)
- [TableName](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dynamodb-table.html#cfn-dynamodb-table-tablename): String
- [Tags](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dynamodb-table.html#cfn-dynamodb-table-tags): [ Tag ]
- [TimeToLiveSpecification](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dynamodb-table.html#cfn-dynamodb-table-timetolivespecification): [TimeToLiveSpecification](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-dynamodb-timetolivespecification.html)
-----
Attributes:
- Arn: String
- StreamArn: String"
`;

exports[`DocumentationService e2e should return docs for sam resources 1`] = `
"[AWS::Serverless::Function](https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction) / Properties


-----
- [Handler](https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction): String
- [Runtime](https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction): String
- [CodeUri](https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction): [CodeUri](https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#s3-location-object)
- [InlineCode](https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction): String
- [FunctionName](https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction): String
- [Description](https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction): String
- [MemorySize](https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction): Integer
- [Timeout](https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction): Integer
- [Role](https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction): String
- [Policies](https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction): Policies
- [Environment](https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction): [Environment](https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#environment-object)
- [VpcConfig](https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction): [VpcConfig](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-lambda-function-vpcconfig.html)
- [Events](https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction): {[key: String]: EventSourceObject}
- [Tags](https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction): {[key: String]: String}
- [Tracing](https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction): String
- [KmsKeyArn](https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction): String
- [DeadLetterQueue](https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#deadletterqueue-object): {[key: String]: DeadLetterQueue}
- [DeploymentPreference](https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#deploymentpreference-object): {[key: String]: DeploymentPreference}
- [Layers](https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction): [ String ]
- [AutoPublishAlias](https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction): String
- [ReservedConcurrentExecutions](https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction): Integer
-----
Attributes:
- Ref: String"
`;

exports[`DocumentationService e2e should return dynamodb resource documentation 1`] = `
"[AWS::DynamoDB::Table](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dynamodb-table.html) / Properties / [KeySchema](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-dynamodb-keyschema.html)


-----
- [AttributeName](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-dynamodb-keyschema.html#aws-properties-dynamodb-keyschema-attributename): String
- [KeyType](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-dynamodb-keyschema.html#aws-properties-dynamodb-keyschema-keytype): String"
`;
